### PR TITLE
[KYUUBI #544] Deregister engine when meeting specified exception stacktraces

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Involve `kyuubi.engine.deregister.exception.stacktraces` to deregister engine when meeting specified exception stacktraces.
### _How was this patch tested?_
Existing UT.

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
Could you please rebase on master to resolve the conflicts?

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/kyuubi/SparkSQLEngineListener.scala [hunk 1]
```diff
@@ -51,6 +52,10 @@ class SparkSQLEngineListener(server: Serverable) extends SparkListener with Logg
        } else if (e.getMessage != null && deregisterMessages.exists(e.getMessage.contains)) {
          deregisterInfo = Some("Job failed exception message matches the specified " +
            s"${ENGINE_DEREGISTER_EXCEPTION_MESSAGES.key}, stopping the engine.")
+       } else if (e.getStackTrace != null &&
+         deregisterStacktraces.exists(e.getStackTrace.mkString("\n").contains)) {
+         deregisterInfo = Some("Job failed exception stacktrace matches the specified " +
+           s"${ENGINE_DEREGISTER_EXCEPTION_STACKTRACES.key}, stopping the engine.")
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/kyuubi/SparkSQLEngineListener.scala:58]: stopping -> deregistering

**Review by yaooqinn — APPROVED** [review]

**Review by pan3793 — APPROVED** [review]
LGTM

**Comment by pan3793** [comment]
Thanks, will merge it for v1.2.0

**Pull request closed without merging**